lib/stream/leo-stream.js

```diff
@@ -212,15 +212,15 @@ module.exports = function(configure) {
 
 			args.push(ls.through((obj, done) => {
 				var e;
-				if (opts.autoDetectPayload && obj.payload) {
+				if (opts.autoDetectPayload && (obj.payload || obj.correlation_id)) {
 					e = obj;
 				} else {
 					e = {
 						payload: obj
 					};
 				}
-				e.id = id;
-				e.event = outQueue;
+				e.id = e.id || id;
```

Comment:
These 2 changes make it so that leo.load can support writing to multiple queues when using kinesis or firehose.  S3 is not supported because currently creates files based on the outQueue which isn't dynamic